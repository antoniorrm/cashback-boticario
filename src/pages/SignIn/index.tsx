import React, { FormEvent, useState } from 'react'
import Button from '@material-ui/core/Button'
import CssBaseline from '@material-ui/core/CssBaseline'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import { Grid, Link } from '@material-ui/core'
import { NavLink } from 'react-router-dom'
import { useDispatch } from 'react-redux'

import { signInRequest } from '../../store/modules/auth/actions'
import LogoBoticatio from '../../assets/img/logo-boticario-primary.svg'

const useStyles = makeStyles(theme => ({
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
	},
	form: {
		width: '100%', // Fix IE 11 issue.
		marginTop: theme.spacing(1),
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
	},
	img: {
		width: '80%',
		margin: theme.spacing(1),
		marginBottom: theme.spacing(4),
	},
}))

export default function SignIn() {
	const classes = useStyles()

	const dispatch = useDispatch()

	const [email, setEmail] = useState<string>('')
	const [password, setPassword] = useState<string>('')

	async function handleSignIn(event: FormEvent) {
		event.preventDefault()
		try {
			await dispatch(
				signInRequest({ email, password })
			)
		} catch (error) {
			console.log(error)
		}
	}

	return (
		<Container component='main' maxWidth='xs'>
			<CssBaseline />
			<div className={classes.paper}>
				<img className={classes.img} src={LogoBoticatio} alt='Logo empresa' />
				<Typography component='h1' variant='h5'>
					Acessar Sistema
				</Typography>
				<form className={classes.form} onSubmit={handleSignIn}>
					<TextField
						variant='outlined'
						margin='normal'
						required
						fullWidth
						id='email'
						label='Email'
						name='email'
						autoComplete='email'
						autoFocus
						value={email}
						onChange={e => setEmail(e.target.value)}
					/>
					<TextField
						variant='outlined'
						margin='normal'
						required
						fullWidth
						name='password'
						label='Senha'
						type='password'
						id='password'
						autoComplete='current-password'
						value={password}
						onChange={e => setPassword(e.target.value)}
					/>
					<Grid container direction='row'>
						<Grid item>
							<Link component={NavLink} to='/sign-up' variant='caption'>
								<Typography>NÃ£o tem conta? Cadastre-se</Typography>
							</Link>
						</Grid>
					</Grid>
					<Button
						type='submit'
						fullWidth
						variant='contained'
						color='primary'
						className={classes.submit}
					>
						Acessar
					</Button>
				</form>
			</div>
		</Container>
	)
}
