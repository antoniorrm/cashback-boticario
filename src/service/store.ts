/* eslint-disable import/prefer-default-export */
import { store } from '../store'

export function getAccessToken() {
	return store.getState().auth.token
}