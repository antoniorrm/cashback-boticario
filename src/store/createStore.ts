import { createStore, applyMiddleware, Middleware, Reducer } from 'redux'
import { PersistState } from 'redux-persist'
import { AuthAction, AuthState } from './modules/auth/types'

export interface StoreState {
	_persist: PersistState
	auth: AuthState
}

export type StoreAction = AuthAction

export default (
	reducers: Reducer<StoreState, StoreAction>,
	middlewares: Middleware[]
) => {
	const enhancer = applyMiddleware(...middlewares)

	return createStore(reducers, enhancer)
}
