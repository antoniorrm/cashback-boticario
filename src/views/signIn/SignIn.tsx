import React from 'react'
import Avatar from '@material-ui/core/Avatar'
import Button from '@material-ui/core/Button'
import CssBaseline from '@material-ui/core/CssBaseline'
import TextField from '@material-ui/core/TextField'
import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import { Grid, Link } from '@material-ui/core'
import { NavLink } from 'react-router-dom'

const useStyles = makeStyles(theme => ({
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
	},
	avatar: {
		margin: theme.spacing(1),
		backgroundColor: theme.palette.secondary.main,
	},
	form: {
		width: '100%', // Fix IE 11 issue.
		marginTop: theme.spacing(1),
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
	},
}))

export default function SignIn() {
	const classes = useStyles()

	return (
		<Container component='main' maxWidth='xs'>
			<CssBaseline />
			<div className={classes.paper}>
				<Avatar className={classes.avatar}>
					<LockOutlinedIcon />
				</Avatar>
				<Typography component='h1' variant='h5'>
					Acessar Sistema
				</Typography>
				<form className={classes.form} noValidate>
					<TextField
						variant='outlined'
						margin='normal'
						required
						fullWidth
						id='email'
						label='Email Address'
						name='email'
						autoComplete='email'
						autoFocus
					/>
					<TextField
						variant='outlined'
						margin='normal'
						required
						fullWidth
						name='password'
						label='Password'
						type='password'
						id='password'
						autoComplete='current-password'
					/>
					<Grid container direction='row'>
						<Grid item>
							<Link component={NavLink} to='/sign-up' variant='caption'>
								<Typography>NÃ£o tem conta? Cadastre-se</Typography>
							</Link>
						</Grid>
					</Grid>
					<Button
						type='submit'
						fullWidth
						variant='contained'
						color='primary'
						className={classes.submit}
					>
						Sign In
					</Button>
				</form>
			</div>
		</Container>
	)
}
