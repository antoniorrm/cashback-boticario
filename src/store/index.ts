import createSagaMiddleware from 'redux-saga'
import { AnyAction, Middleware, Store } from 'redux'
import { persistStore } from 'redux-persist'

import createStore from './createStore'
import rootReducer from './modules/rootReducer'
import rootSaga from './modules/rootSaga'
import persistReducers from './persistedReducers'

const sagaMiddleware = createSagaMiddleware()

const middlewares: Middleware[] = [sagaMiddleware]

const store = createStore(persistReducers(rootReducer), middlewares)
const persistor = persistStore(store as unknown as Store<any, AnyAction>)

sagaMiddleware.run(rootSaga)

export { store, persistor }
