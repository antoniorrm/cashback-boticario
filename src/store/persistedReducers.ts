import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import { Reducer } from 'redux'
import { StoreAction, StoreState } from './createStore'

export default (reducers: Reducer<StoreState, StoreAction>) => {
	const persistedReducers = persistReducer(
		{
			key: 'cashback',
			storage,
			whitelist: ['auth'],
		},
		reducers
	)

	return persistedReducers
}
