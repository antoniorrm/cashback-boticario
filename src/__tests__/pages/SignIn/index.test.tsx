/* eslint-disable no-undef */
import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { Provider } from 'react-redux'
import * as reactRedux from 'react-redux'
import { Router } from 'react-router-dom'
import { SnackbarProvider } from 'notistack'
// import MockAdapter from 'axios-mock-adapter'

import SignIn from '../../../pages/SignIn'
import { store } from '../../../store'
// import api from '../../../service/api'
// const apiMock = new MockAdapter(api)

describe('Login page render sucessfully', () => {
	it('renders component correctly', () => {
		const history = createMemoryHistory()
		const { container } = render(
			<Provider store={store}>
				<Router history={history}>
					<SnackbarProvider maxSnack={1}>
						<SignIn />
					</SnackbarProvider>
				</Router>
			</Provider>
		)
		expect(container).toMatchSnapshot()
	})
	it('should be Button click in component', () => {
		const history = createMemoryHistory()
		const mockDispatch = jest.fn()
		const useDispatch = jest.spyOn(reactRedux, 'useDispatch')
		useDispatch.mockReturnValue(mockDispatch)
		render(
			<Provider store={store}>
				<Router history={history}>
					<SnackbarProvider maxSnack={1}>
						<SignIn />
					</SnackbarProvider>
				</Router>
			</Provider>
		)
		const Button = screen.getByRole('button', { name: 'Acessar' })
		userEvent.click(Button)
		expect(mockDispatch).toHaveBeenCalled()
	})
})
