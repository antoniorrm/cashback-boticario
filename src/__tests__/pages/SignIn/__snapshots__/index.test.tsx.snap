// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page render sucessfully renders component correctly 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-1"
    >
      <img
        alt="Logo empresa"
        class="makeStyles-img-4"
        src="logo-boticario-primary.svg"
      />
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Acessar Sistema
      </h1>
      <form
        class="makeStyles-form-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
            for="email"
            id="email-label"
          >
            Email
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="email"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="email"
              name="email"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
              >
                <span>
                  Email
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            Senha
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7"
              >
                <span>
                  Senha
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-caption MuiTypography-colorPrimary"
              href="/sign-up"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Não tem conta? Cadastre-se
              </p>
            </a>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Acessar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</div>
`;
